### PR TITLE
show locals in the debugger

### PR DESCRIPTION
**Pull request by devoncarew**
- create a new `MTree` component
- use it to show local variables in the execution view
- extend the debug model to know about values (`DebugValue`)

@danrubel

### CODE REVIEW

**Comment by devoncarew** [comment]
screen shot 2015-12-07 at 11 02 01 am

File: lib/debug/debugger_ui.dart [hunk 1]
```diff
@@ -391,16 +393,58 @@ class ExecutionTab extends MTab {
...
+      locals.update([]);
+      return;
+    }
 
     frame.location.resolve().then((DebugLocation location) {
       if (location.resolvedPath) view._jumpToLocation(location);
     });
+
+    List<DebugVariable> vars = frame.locals;
+    locals.update(vars == null ? [] : vars);
```
Comment (danrubel) [inline lib/debug/debugger_ui.dart]: Shorten to this? ... `locals.update(vars ?? []);`

File: lib/debug/observatory_debugger.dart [hunk 2]
```diff
@@ -521,13 +527,108 @@ class ObservatoryFrame extends DebugFrame {
...
+        return [];
+      }
+    });
+  }
+
+  // TODO: handle truncated
+  String get valueAsString => value.valueAsString;
+}
+
+class SentenalValue extends DebugValue {
```
Comment (danrubel) [inline lib/debug/observatory_debugger.dart]: SentenalValue --> SentinalValue

**Comment by danrubel** [comment]
LGTM

File: lib/debug/observatory_debugger.dart (hunk 2 above)
Comment (devoncarew) [inline lib/debug/observatory_debugger.dart]: oh dear

File: lib/debug/debugger_ui.dart (hunk 1 above)
Comment (devoncarew) [inline lib/debug/debugger_ui.dart]: done

**Pull request merged**